helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -91,6 +160,52 @@ public RoutingTableProvider() {
     return instanceSet;
   }
 
+  /**
+   * returns all instances for all resources from {resource} group that are in a specific {state}
+   *
+   * @param resource
+   * @param state
+   * @param groupingEnabled
+   *
+   * @return empty list if there is no instance in a given state
+   */
+  public Set<InstanceConfig> getInstances(String resource, String state, boolean groupingEnabled) {
```

Comment:
Also, as @kishoreg mentioned above, it's better to have two methods rather than this boolean flag.